Move workflow utils into rose apps

jfrost-mo opened the pull request:
As the workflow is now inside the package, its scripts are able to be tested in place.

The simple, self-contained scripts were just moved over, while the more complex fetch_data script needed to use relative imports to make the fetch_data infrastructure available to the other scripts, given they are in the same directory.

While the scripts where being moved I took the opportunity to add licence headers where they were missing.

The removes the last file in _workflow_utils, so that subpackage is deleted.

Based on #1474.

<!-- Thanks for contributing! Please add a short description of your change, and link to an issue, e.g. "Fixes #123" -->

### Contribution checklist

Aim to have all relevant checks ticked off before merging. See the [developer's guide](https://metoffice.github.io/CSET/contributing/) for more detail.

- [x] Documentation has been updated to reflect change.
- [x] New code has tests, and affected old tests have been updated.
- [x] All tests and CI checks pass.
- [x] Ensured the pull request title is descriptive.
- [ ] Conda lock files have been updated if dependencies have changed.
- [ ] Attributed any Generative AI, such as GitHub Copilot, used in this PR.
- [x] Marked the PR as ready to review.

ukmo-huw-lewis reviewed (approved):
Happy with proposed changes, and content all pytest functions succeeding. 

Observe pushing the python into relevant app/*/bin/ directories complicates the calling path in e.g. pytest, but I think clarifies what sits where from user and developer perspectives (and reduced number of files in code base I guess??).

Beyond pytest, can I check if you also ran a full cset-workflow test based on this branch? If so, useful to point to output path. Given frequency of connected PR, I'm also content if we run a clean cset-workflow test after merging recent PR.

Marking as Approved. To be merged following #1474 and #1487.

jfrost-mo commented:
I haven't yet run a full CSET workflow. I'll do that on this branch ahead of merging it and #1474.

jfrost-mo commented:
Testing revealed issues with the relative importing of the fetch_data script. I've converted them to absolute imports that rely on the current working directory being added to the `PYTHONPATH`. The means that the other fetch-data scripts can't be tested, but otherwise should work.

I've also fixed a couple other things I missed, such the shebang in preprocess or the `if __main__ == "__name__"` in finish_website.
